src/Storages/MergeTree/MergeTreeMetadataCache.h

```diff
@@ -0,0 +1,41 @@
+#pragma once
+
+#include "config_core.h"
+
+#if USE_ROCKSDB
+#include <base/types.h>
+#include <Core/Types.h>
+#include <Poco/Logger.h>
+#include <rocksdb/table.h>
+#include <rocksdb/db.h>
+
+namespace DB
+{
+class MergeTreeMetadataCache
+{
+public:
+    using Status = rocksdb::Status;
+
+    explicit MergeTreeMetadataCache(rocksdb::DB * rocksdb_) : rocksdb{rocksdb_}
+    {
+        assert(rocksdb);
+    }
+
+    MergeTreeMetadataCache(const MergeTreeMetadataCache &) = delete;
+    MergeTreeMetadataCache & operator=(const MergeTreeMetadataCache &) = delete;
+
+    Status put(const String & key, const String & value);
+    Status del(const String & key);
+    Status get(const String & key, String & value);
+    void getByPrefix(const String & prefix, Strings & keys, Strings & values);
+
+    void shutdown();
+private:
+    std::unique_ptr<rocksdb::DB> rocksdb;
```

Comment:
In addition to `MergeTreeMetadataCache*` metrics, worth export size of this cache in https://github.com/ClickHouse/ClickHouse/blob/9b3493ba9a580c7b6c418fc1d4f9c8405e65e882/src/Interpreters/AsynchronousMetrics.cpp#L537-L543

Something like this:

```c++
uint64_t keys = 0;
rocksdb->GetAggregatedIntProperty("rocksdb.estimate-num-keys", &keys);
```

Reply:
done